test/e2e/cluster/add_remove_cluster.go

```diff
@@ -0,0 +1,52 @@
+package cluster
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/submariner-io/submariner/test/e2e/dataplane"
+	"github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var _ = Describe("[expansion] Test expanding/shrinking an existing cluster fleet", func() {
+	f := framework.NewDefaultFramework("add-remove-cluster")
+
+	It("Should be able to add and remove third cluster", func() {
+		clusterAName := framework.TestContext.KubeContexts[framework.ClusterA]
+		clusterBName := framework.TestContext.KubeContexts[framework.ClusterB]
+		clusterCName := framework.TestContext.KubeContexts[framework.ClusterC]
+
+		gatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, true)
+		Expect(gatewayNode).To(HaveLen(0), fmt.Sprintf("Expected no gateway node on %q", framework.ClusterC))
+
+		By(fmt.Sprintf("Verifying that a pod in cluster %q cannot connect to a pod in cluster %q", clusterAName, clusterCName))
+		dataplane.RunNoConnectivityTest(f, false, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterA, framework.ClusterC)
+
+		By(fmt.Sprintf("Verifying that a pod in cluster %q cannot connect to a service in cluster %q", clusterBName, clusterCName))
+		dataplane.RunNoConnectivityTest(f, true, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterB, framework.ClusterC)
+
+		nonGatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, false)
+		Expect(nonGatewayNode).To(HaveLen(1), fmt.Sprintf("Found non-gateway node %q on %q", nonGatewayNode[0].Name, clusterCName))
```

Comment:
_cluster3_ is setup to have 2 worker nodes.  But you don't care exactly how many there are only that it's non-zero.

```suggestion
		Expect(nonGatewayNodes).ToNot(HaveLen(0), fmt.Sprintf("Expected at least 1 gateway node on %q", framework.ClusterC))
```